app/lib/features/chat_ng/widgets/reply_preview.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:acter/features/chat_ng/models/message_metadata.dart';
+import 'package:acter/features/chat_ng/models/reply_message_state.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:acter/features/chat_ng/widgets/events/reply_original_event.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:acter/common/extensions/options.dart';
+
+// Reply State UI widget
+class ReplyPreview extends ConsumerWidget {
+  final MessageMetadata metadata;
+  const ReplyPreview({
+    super.key,
+    required this.metadata,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final String roomId = metadata.roomId;
+    final String messageId = metadata.messageId;
+    final String originalId =
+        metadata.repliedTo.expect('should always contain replied id');
+    final ReplyMsgInfo replyInfo =
+        (roomId: roomId, messageId: messageId, originalId: originalId);
+
+    final roomMsgState = ref.watch(replyToMsgProvider(replyInfo)).valueOrNull;
+
+    return switch (roomMsgState) {
+      ReplyMsgLoading() => replyBuilder(
+          context,
+          Center(child: CircularProgressIndicator()),
```

Comment:
please let's not use any circular progress indicators going forward. They make for a messy and constantly switching UI. Let's use skeletons!